static/js/message_viewport.js

```diff
@@ -41,14 +44,14 @@ exports.message_viewport_info = function () {
 };
 
 exports.at_bottom = function () {
-    const bottom = exports.scrollTop() + exports.height();
-    const full_height = exports.message_pane.prop("scrollHeight");
+    const bottom = exports.scrollTop() + window.innerHeight;
+    const full_height = document.body.scrollHeight;
 
     // We only know within a pixel or two if we're
     // exactly at the bottom, due to browser quirkiness,
     // and we err on the side of saying that we are at
     // the bottom.
-    return bottom + 2 >= full_height;
+    return bottom + 5 >= full_height;
```

Comment:
Hmm, what happened here?

Reply:
I think this a bug unrelated to this PR where the current logic does not take into account the `#panels` element.
Thus the `message_viewport.at_bottom` always returns `false` when the panel is displayed.
The bugs due to this are also reproducible on CZO.

For example, using the `PageDown` key, the condition on [this line](https://github.com/zulip/zulip/blob/master/static/js/navigate.js#L113) does not satisfy.
Thus in the gif, the 1st key press causes the scrollbar to shift down and the 2nd key press makes the pointer select the bottom most message which should have been done by the first key press.
On removing the panel, the behaviour gets fixed.
![PgDn](https://user-images.githubusercontent.com/40122794/95354429-0be32a00-08e2-11eb-962c-cebb13f3a847.gif)

There is also a gap below the sidebars if the page is loaded with the panel and then removed.
The changes in this commit fixes it.

I'll revert this change and test this commit without the panel so that behaviour remains the same.
And create another commit for the changes when panel is displayed.